streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -877,23 +881,46 @@ private void connectProcessorAndStateStore(String processorName, String stateSto
         return sourceTopics;
     }
 
-    private void connectStateStoreNameToSourceTopics(final String stateStoreName,
-                                                     final ProcessorNodeFactory processorNodeFactory) {
+    private Set<Pattern> findSourcePatternsForProcessorParents(String[] parents) {
```

Comment:
nit: add `final` to parameter (there are some more like this -- please update all).